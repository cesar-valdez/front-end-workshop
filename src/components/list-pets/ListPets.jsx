import React from 'react';
import PropTypes from 'prop-types';
import PetCard, { PetCardSchema } from '../pet-card';

import './list-pets.scss';

const ListPets = ({ pets }) => (
  <div className="yta_ListPets">
    {pets.map(pet => (
      <PetCard key={pet.id} {...pet} />
    ))}
  </div>
);

ListPets.prototypes = {
  pets: PropTypes.arrayOf(PetCardSchema).isRequired,
};

export default ListPets;
