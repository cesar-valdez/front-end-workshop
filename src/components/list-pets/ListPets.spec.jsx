import React from 'react';
import { shallow } from 'enzyme';

import Pets from '../../__mocks__/Pets';
import ListPets from './ListPets';

describe('ListPets', () => {
  const listPets = shallow(<ListPets pets={Pets} />);

  it('should render LisPets compnente', () => {
    expect(listPets).toMatchSnapshot();
  });
});
