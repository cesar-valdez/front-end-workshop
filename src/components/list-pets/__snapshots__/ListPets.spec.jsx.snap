// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPets should render LisPets compnente 1`] = `
<div
  className="yta_ListPets"
>
  <PetCard
    details="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad eum dolorum architecto obcaecati enim dicta praesentium, quam nobis! Neque ad aliquam facilis numquam. Veritatis, sit."
    id="abc"
    img="https://duaw26jehqd4r.cloudfront.net/items/1M0C3r3X2t0x2q342A2A/benito.png?v=933367f2"
    key="abc"
    name="Benito"
  />
  <PetCard
    details="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad eum dolorum architecto obcaecati enim dicta praesentium, quam nobis! Neque ad aliquam facilis numquam. Veritatis, sit."
    id="def"
    img="https://duaw26jehqd4r.cloudfront.net/items/0O3b091L2O3N3x1l2Q3Q/remmy.png?v=69c651fc"
    key="def"
    name="Remmy"
  />
</div>
`;
